lib/auth/usertoken.go

```diff
@@ -452,6 +465,110 @@ func (s *Server) createRecoveryToken(ctx context.Context, username, tokenType st
 	return newToken, nil
 }
 
+// CreatePrivilegeToken implements AuthService.CreatePrivilegeToken.
+func (s *Server) CreatePrivilegeToken(ctx context.Context, req *proto.CreatePrivilegeTokenRequest) (types.UserToken, error) {
+	username, err := GetClientUsername(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	authPref, err := s.GetAuthPreference(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if !authPref.GetAllowLocalAuth() {
+		return nil, trace.AccessDenied("local auth needs to be enabled")
+	}
+
+	secondFactor := authPref.GetSecondFactor()
+	if secondFactor == constants.SecondFactorOff {
+		return nil, trace.AccessDenied("second factor must be enabled")
+	}
+
+	tokenKind := UserTokenTypePrivilege
+
+	// Begin authenticating second factor.
+	switch req.GetExistingMFAResponse().GetResponse().(type) {
+	case nil:
```

Comment:
Unnecessary type switch?

```suggestion
	switch {
	case req.GetExistingMFAResponse().GetResponse() == nil:
```